<?xml version="1.0"?>
<launch>

  ############################# Global Param ###################################

  <!--   flight_mode
      *pos mode: pos control with hokuyo + keyboard,
      *vel mode: vel control with px4 + joystick
      the most important param
      controlMode: switch betwen vel and pos
      altitudeControlMode: switch between laser and sonoar
      usefreefall: necessary for snoar mode
      useouteryawest: necessary for lasersalm + pos
  -->

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />


  ############################# JSK Quadcopter ###################################

  <node pkg="aerial_robot_base" type="aerial_robot_base_node" name="di_control_node"  launch-prefix="$(arg launch_prefix)" output="screen" > 

    ############### Basic Param  #####################
    <param name="rx_loop_rate" type="double" value="110"/>
    <param name="tx_loop_rate" type="double" value="40"/>
    <param name="tf_pub_loop_rate" type="double" value="120"/>

    <param name="motor_num" type="int" value="1"/>

    <param name="estimator/altitude_control_mode" type="int" value="0"/>
    <param name="estimator/use_outer_yaw_est" type="bool" value="true"/>
    <!-- x/y axis -> pos control: 0, vel control: 1 -->
    <param name="navigator/xy_control_mode" type="int" value="0"/>

    <!--degub -->
    <param name="navigator/gain_tunning_mode" type="int" value="0"/>


    ############### Quadctoper Basic State ######################
    <rosparam file="$(find di_control)/config/EstimatorConfig.yaml" command="load" />
    ############### PID Control ######################
    <rosparam file="$(find di_control)/config/PidControlConfig.yaml" command="load" />

    ############### _motion Capture####################
    <param name="mocap/cog_offset" type="bool" value="false"/><!-- new for new hydra control -->
    <param name="estimator/mocap_flag" type="bool" value="true"/>
    <param name="estimator/px4flow_flag" type="bool" value="false"/>

    ############### Kalman Filter ####################
    <param name="estimator/kalman_filter_flag" type="bool" value="false"/>

    ############### Keyboard Navigation ##############
    <rosparam file="$(find di_control)/config/TeleopNavigationConfig.yaml" command="load" />

    ############### remap topic ######################
    ### Optical Flow ##############
    <remap from="optical_flow/opt_flow" to="px4flow/opt_flow" />
    ### Joy Stick #################
    <remap from="/navigator/joy_stick_command" to="/joy" />
    ### Teleop Navigator ##########
    <remap from="/navigator/teleop_command/land" to="/teleop_command/land" />
    <remap from="/navigator/teleop_command/start" to="/teleop_command/start" />
    <remap from="/navigator/teleop_command/halt" to="/teleop_command/halt" />
    <remap from="/navigator/teleop_command/takeoff" to="/teleop_command/takeoff" />
    <remap from="/navigator/teleop_command/roll" to="/teleop_command/roll" />
    <remap from="/navigator/teleop_command/pitch" to="/teleop_command/pitch" />
    <remap from="/navigator/teleop_command/yaw" to="/teleop_command/yaw" />
    <remap from="/navigator/teleop_command/throttle" to="/teleop_command/throttle" />
    <remap from="/navigator/teleop_command/ctrl_mode" to="/teleop_command/ctrl_mode" />

    <remap from="/arming_ack" to="/jsk_stm_serial_comm/arming_ack" />
    <remap from="/aerial_robot_control" to="/jsk_stm_serial_comm/aerial_robot_control" />
    <remap from="/config_cmd" to="/jsk_stm_serial_comm/config_cmd" />


    <remap from="/kduino/rc_cmd" to="/deprected" />

<!--    <remap from="/controller/debug" to="/test/controller/debug" /> -->
<!--
    <remap from="/imu/kduino/imu" to="/kduino/imu" />
    <remap from="/navigator/kduino/msp_cmd" to="/kduino/msp_cmd" />
-->

  </node>

  ############### Motion Capture####################


  ################################# jsk_stm ########################################
<include file="$(find aerial_robot_base)/launch/sensors/d_board.launch" >
  <arg name="port" value="/dev/ttyUSB1" />
</include>

  ################################# Joy Stick ######################################


<!--
  <include file="$(find aerial_robot_base)/launch/sensors/joy_stick.launch" />
-->


################################## dynamixel motor #########################
<include file="$(find di_control)/launch/servo/servo.launch" />

</launch>

